protocol/msgpack/options.go

```diff
@@ -21,25 +21,25 @@
 package msgpack
 
 import (
-	"errors"
-
 	"github.com/m3db/m3metrics/pool"
 	xpool "github.com/m3db/m3x/pool"
 )
 
 const (
-	defaultIgnoreHigherVersion = true
-)
+	// Whether the iterator should ignore higher-than-supported version
+	// by default for unaggregated metrics
+	defaultUnaggregatedIgnoreHigherVersion = true
 
-var (
-	errNoFloatsPool   = errors.New("no floats pool")
-	errNoPoliciesPool = errors.New("no policies pool")
+	// Whether the iterator should ignore higher-than-supported version
+	// by default for aggregated metrics
+	defaultAggregatedIgnoreHigherVersion = true
```

Comment:
I'd argue fairly strongly these defaults should be false.  In production you want to always know you are dropping data/not decoding data.